### DISCUSSION TITLE
Get API call result Displayed on a Budibase Form

### DISCUSSION THREAD

**Discussion by SaraFattouh** [Budibase/budibase#9315] [Help]
**Describe the feature request**
A clear and concise description of what the feature request.

**Screenshots**

**Comment by SaraFattouh** [comment]
I've been learning and testing Budibase platform and I am impressed by the REST Source features that you provide. 
I have a sample API call that retrieve people data from [SWAPI API ] 
I am using binding in my form to read the param which is the person id. 

https://swapi.dev/api/people?id=idValue

and I am receiving the results in data [which is accessible in the transformer of the request]. 

Is there a way to display the query results in the form, for example binding the data to a component feature ? 

Any suggestion is highly appreciated! 

<img width="1511" alt="Screenshot 2023-01-11 at 16 42 41" src="https://user-images.githubusercontent.com/31407456/211851330-8bb87e14-c099-4097-b50d-bbef25fe22d2.png">
<img width="1505" alt="Screenshot 2023-01-11 at 16 48 37" src="https://user-images.githubusercontent.com/31407456/211851847-dde83d6c-2013-4081-b0a5-51cd1effa994.png">
<img width="1512" alt="Screenshot 2023-01-11 at 16 50 42" src="https://user-images.githubusercontent.com/31407456/211852364-266a30b5-f3f8-4153-84e3-791005c95f3e.png">

**Reply by melohagan** [reply]
Hey @SaraFattouh 

Apologies for the late response!

That looks good so far, I'm assuming after you **Execute Query** you want to display that response data in the paragraph component?
You could do that by using [app state](https://docs.budibase.com/docs/app-state) and [action referencing](https://docs.budibase.com/docs/chaining-actions#action-referencing) 

So you actions would look like:
1. Execute Query
2. Update State ( Bind some key to: `{{ Action 1.Query Result.data.0.key }}` )

Then in your paragraph component, you would bind the value: `{{ State.key }}`
